File: 백준/Silver 5/1094.py
X = int(input())
# 막대는 64의 약수로만 나눠진다.
bars = [1,2,4,8,16,32,64]
# X와 값을 비교할 변수 생성
total = 0
# 갯수 세는 변수 생성
cnt = 0
# 리스트 거꾸로 출력
for i in bars[::-1]:
    # 만약 X가 i 보다 큰 경우
    if X > i:
        #값을 더해준다
        total += i
        # 값이 X보다 큰 경우
        if total > X:
            # 다시 값을 뺸다
            total -= i
        # 그 외의 경우 cnt를 증가시킨다.
        else:
            cnt += 1
    # 만약 X 가 i 랑 같은 경우 즉 리스트 값인경우 break를 시켜준다
    elif X == i:
        cnt += 1
        break
print(cnt)

File: JavaScript/백준/Silver 5/1094.js
// 인풋
const fs = require("fs");
const input = parseInt(fs.readFileSync("./1094.txt").toString().trim());
// 막대길이 64
let bar = 64;
// 자른 막대 저장 리스트
let barList = [];
// 총 길이 체크
let total = 0;
// 몇개 사용했는지 체크
let count = 0;
if (input === bar) {
  // 막대가 64이고 X값이 64인 경우 1개의 막대만 사용
  console.log(count + 1);
} else {
  // 외의 경우 막대가 1이 나올때 까지 분리
  while (bar > 1) {
    // check 변수를 통해서 X 보다 작은 막대를 다 저장
    const check = bar / 2;
    if (check <= input) {
      barList.push(check);
    }
    bar /= 2;
  }
  //   저장된 막대를 더하고 빼면서 몇개를 사용했는지 체크
  for (bar of barList) {
    total += bar;
    count += 1;
    if (total > input) {
      total -= bar;
      count -= 1;
    } else if (total == input) {
      console.log(count);
      break;
    }
  }
}
